### PR TITLE
Feature - TriNguyen - CollectionView

### PR DESCRIPTION
**Pull request by mvn-tringuyen-dn**

### CODE REVIEW

File: 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex4/HeaderCell/HeaderCell.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        collectionView.collectionViewLayout = layout
+    }
+    
+    // MARK: - IBActions
+    @IBAction private func previousButtonTouchUpInside(_ sender: Any) {
+        currentIndex.row -= 1
+        if currentIndex.row == 0 { return }
```
Comment (mvn-tanhuynh-dn) [inline 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex4/HeaderCell/HeaderCell.swift:43]: Why return if currentIndex.row == 0?

File: 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex4/ItemCell/Ex4CollectionCell.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        return UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+    }
+}
+
+// MARK: - Config
+extension Ex4CollectionCell {
+    struct Config {
```
Comment (mvn-tanhuynh-dn) [inline 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex4/ItemCell/Ex4CollectionCell.swift:57]: Add space

File: 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex5/Ex5ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        return 5
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.row == 0 {
+            let cell = tableView.dequeueReusableCell(withIdentifier: Identifier.slideCell.rawValue, for: indexPath) as! SlideTableViewCell
+            return cell
+        } else {
+            let cell = tableView.dequeueReusableCell(withIdentifier: Identifier.ex5Cell.rawValue, for: indexPath) as! Ex5TableViewCell
+            return cell
```
Comment (mvn-tanhuynh-dn) [inline 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex5/Ex5ViewController.swift:59]: Don't force upwrap "!" optional value, please use "?" instead

File: 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex5/SlideCell/SlideTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    // MARK: - Outlet
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var previousButton: UIButton!
+    @IBOutlet private weak var nextButton: UIButton!
+    
+    // MARK: - Properties
+    private var currentIndex: IndexPath = IndexPath(row: 0, section: 0) {
+        didSet {
+            currentIndex.row = currentIndex.row < 0 ? 0 : currentIndex.row
+            currentIndex.row = currentIndex.row > 11 ? 11 : currentIndex.row
```
Comment (mvn-tanhuynh-dn) [inline 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex5/SlideCell/SlideTableViewCell.swift:21]: Group two line into one line:
currentIndex.row = (currentIndex.row < 0) ? 0 : (currentIndex.row > 11 ? 11 : currentIndex.row)

File: 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex4/HeaderCell/HeaderCell.swift (hunk 1 above)
Comment (mvn-tringuyen-dn) [reply 12_CollectionView/TriNC/CollectionView/CollectionView/AppDelegate/Scene/Ex4/HeaderCell/HeaderCell.swift:43]: Running the program crashed so check currentindex forgot to remove that condition.

**Review by mvn-tanhuynh-dn — APPROVED** [review]
LGTM!

**Pull request merged**